### PR TITLE
Run Syncthing with ionice (fixes #299).

### LINKED ISSUE #299: Syncthing freezes during initial scan of large folders
I have a large folder (~32 GB, 10^4 files) I added. The GUI freezes during the scanning process, and the only way to get it to update is to Force Close and restart the app.

### PR DESCRIPTION
**Pull request by Zillode**
(untested)
@ST_Jenkins test this please

### CODE REVIEW

**Comment by Zillode** [comment]
test this please

**Comment by Zillode** [comment]
test this please

**Comment by Zillode** [comment]
test this please

File: src/main/java/com/nutomic/syncthingandroid/syncthing/SyncthingRunnable.java [hunk 1]
```diff
@@ -84,6 +87,45 @@ public void run() {
         }
     }
 
+    private void niceSyncthing() {
```
Comment (Nutomic) [inline src/main/java/com/nutomic/syncthingandroid/syncthing/SyncthingRunnable.java]: Can you put a javadoc here please?

File: src/main/java/com/nutomic/syncthingandroid/syncthing/SyncthingRunnable.java [hunk 2]
```diff
@@ -84,6 +87,45 @@ public void run() {
...
+        new Thread() {
+            public void run() {
+                Process niceST = null;
+                DataOutputStream niceSTOut = null;
```
Comment (Nutomic) [inline src/main/java/com/nutomic/syncthingandroid/syncthing/SyncthingRunnable.java]: `niceSTOut` kinda sucks for a camelCase var name, should be `niceStOut` or better just `niceOut`.

File: src/main/java/com/nutomic/syncthingandroid/syncthing/SyncthingRunnable.java [hunk 3]
```diff
@@ -84,6 +87,45 @@ public void run() {
...
+                int ret = 1;
+                try {
+                    Thread.sleep(1000); // Wait a second before getting the Pid
+                    niceST = Runtime.getRuntime().exec("sh");
+                    niceSTOut = new DataOutputStream(niceST.getOutputStream());
+                    niceSTOut.writeBytes("set `ps |grep libsyncthing.so`\n");
+                    niceSTOut.writeBytes("ionice $2 be 7\n"); // best-effort, low priority
+                    niceSTOut.writeBytes("exit\n");
+                    log(niceST.getInputStream(), Log.INFO);
```
Comment (Nutomic) [inline src/main/java/com/nutomic/syncthingandroid/syncthing/SyncthingRunnable.java]: Not sure if we need this as we also log exceptions etc. It also gets lumped in with syncthing logs. So I'd use a different tag or just don't log the output.

Comment (Zillode) [inline src/main/java/com/nutomic/syncthingandroid/syncthing/SyncthingRunnable.java]: - Only log warnings (should be empty)
- Use different tag

**Comment by Zillode** [comment]
Bump

**Pull request merged**